scapy/supersocket.py

```diff
@@ -32,7 +32,14 @@ def send(self, x):
         sx = str(x)
         if hasattr(x, "sent_time"):
             x.sent_time = time.time()
-        return self.outs.send(sx)
+        try:
+            return self.outs.send(sx)
+        except socket.error, msg:
+            if msg[0] == 22 and len(sx) < conf.min_pkt_size:
```

Comment:
As this is Linux specific, I think that this patch must be moved to scapy/arch/linux.py